### PR TITLE
[pallet-revive] code size API

### PR DESCRIPTION
**Pull request by xermicus**
This PR implements the contract API to query the code size of a given address.

### CODE REVIEW

**Comment by xermicus** [comment]
bot fmt

**Comment by xermicus** [comment]
bot bench substrate-pallet --pallet=pallet_revive

File: substrate/frame/revive/src/wasm/runtime.rs [hunk 1]
```diff
@@ -1414,6 +1417,22 @@ pub mod env {
 		)?)
 	}
 
+	/// Retrieve the code size for a given contract address.
+	/// See [`pallet_revive_uapi::HostFn::code_size`].
+	#[api_version(0)]
+	fn code_size(&mut self, memory: &mut M, addr_ptr: u32, out_ptr: u32) -> Result<(), TrapReason> {
+		self.charge_gas(RuntimeCosts::CodeSize)?;
+		let mut address = H160::zero();
+		memory.read_into_buf(addr_ptr, address.as_bytes_mut())?;
```
Comment (athei) [inline substrate/frame/revive/src/wasm/runtime.rs:1426]: Can you add a `read_h160` function analog to `read_h256` as drive by?

**Review by athei — APPROVED** [review]

File: substrate/frame/revive/src/exec.rs [hunk 2]
```diff
@@ -1562,6 +1565,13 @@ where
 			})
 	}
 
+	fn code_size(&self, address: &H160) -> U256 {
+		<ContractInfoOf<T>>::get(&address)
+			.and_then(|contract| CodeInfoOf::<T>::get(contract.code_hash))
+			.map(|info| info.code_len())
+			.unwrap_or(U256::zero())
```
Comment (pgherveou) [inline substrate/frame/revive/src/exec.rs:1572]:
```suggestion
			.unwrap_or_default()
```

**Review by pgherveou — APPROVED** [review]

**Comment by xermicus** [comment]
bot bench substrate-pallet --pallet=pallet_revive

**Pull request merged**